### PR TITLE
refactor: brand component

### LINKED ISSUE #1574: Header: why is the logo always supposed to get a link?
Currently there's no way around for the logo within `DB Brand` to have a link, even if none is provided by the property there's still the `href` of `/` being enforced. Is this a conceptual requirement, or could we switch back to having this optional for the developers, depending on their needs and requirements?

### PR DESCRIPTION
**Pull request by nmerget**
## Proposed changes

closes #1574

- reworked brand completely 

## Types of changes

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [x] Refactoring (fix on existing components or architectural decisions)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation Update (if none of the other choices apply)

### CODE REVIEW

**Comment by mfranzke** [comment]
set to draft as the pipeline fails.

File: showcases/angular-showcase/src/app/components/brand/brand.component.html [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	<ng-template
+		#exampleTemplate
+		let-exampleProps="exampleProps"
+		let-exampleName="exampleName"
+		let-exampleIndex="exampleIndex"
+		let-variantIndex="variantIndex"
+	>
+		<db-brand [hideLogo]="exampleProps.hideLogo">
+			<img
+				*ngIf="exampleProps.customLogo"
```
Comment (d-koppenhagen) [inline showcases/angular-showcase/src/app/components/brand/brand.component.html:15]: Should be refactored to use Control Flow here

File: showcases/angular-showcase/src/app/components/brand/brand.component.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { Component } from '@angular/core';
+import defaultComponentVariants from '../../../../../shared/brand.json';
+
+@Component({
+	selector: 'app-button',
+	templateUrl: './brand.component.html'
```
Comment (d-koppenhagen) [inline showcases/angular-showcase/src/app/components/brand/brand.component.ts:6]:
```suggestion
	templateUrl: './brand.component.html',
	standalone: true,
```

File: showcases/angular-showcase/src/app/app.module.ts [hunk 3]
```diff
@@ -60,11 +60,13 @@ import { NavItemComponent } from './nav-item/nav-item.component';
 import { MainNavigationComponent } from './components/main-navigation/main-navigation.component';
 import { TextareaComponent } from './components/textarea/textarea.component';
 import { IconComponent } from './components/icon/icon.component';
+import { BrandComponent } from './components/brand/brand.component';
 
 @NgModule({
```
Comment (d-koppenhagen) [inline showcases/angular-showcase/src/app/app.module.ts:65]: NgModule was removed. Component must be imported directly in the parent Standalone Component

File: showcases/angular-showcase/src/app/components/brand/brand.component.html (hunk 1 above)
Comment (nmerget) [reply showcases/angular-showcase/src/app/components/brand/brand.component.html]: fixed

File: showcases/angular-showcase/src/app/components/brand/brand.component.ts (hunk 2 above)
Comment (nmerget) [reply showcases/angular-showcase/src/app/components/brand/brand.component.ts]: fixed

File: showcases/angular-showcase/src/app/app.module.ts (hunk 3 above)
Comment (nmerget) [reply showcases/angular-showcase/src/app/app.module.ts]: fixed

File: __snapshots__/brand/showcase/webkit/regular/neutral/DBBrand-should-match-screenshot.png
Comment (mfranzke) [inline __snapshots__/brand/showcase/webkit/regular/neutral/DBBrand-should-match-screenshot.png:1]: is "Custom Logo" with a logo included like the others actually something that our users would understand?

File: showcases/react-showcase/src/components/brand/index.tsx [hunk 4]
```diff
@@ -1,39 +1,22 @@
...
+	hideLogo,
+	customLogo
+}: DBBrandProps & { customLogo: boolean }) => (
+	<DBBrand hideLogo={hideLogo}>
+		{customLogo && (
+			<img
+				src={`${
+					process?.env?.NEXT_PUBLIC_BASE_PATH ?? '/react-showcase'
+				}/assets/images/db_logo.svg`}
+			/>
```
Comment (mfranzke) [inline showcases/react-showcase/src/components/brand/index.tsx:18]:
```suggestion
				}/assets/images/db_logo.svg`}
				alt=""
			/>
```

Comment (nmerget) [reply showcases/react-showcase/src/components/brand/index.tsx]: fixed

File: showcases/vue-showcase/src/components/brand/Brand.vue [hunk 5]
```diff
@@ -0,0 +1,28 @@
+<script setup lang="ts">
+import DefaultComponent from "../DefaultComponent.vue";
+import defaultComponentVariants from "../../../../shared/brand.json";
+import { DBBrand } from "../../../../../output/vue/src";
+import imageUrl from "/assets/images/placeholder.jpg";
+
+const log = (exampleName: string) => {
+	// eslint-disable-next-line no-alert
+	alert(exampleName);
+};
```
Comment (mfranzke) [inline showcases/vue-showcase/src/components/brand/Brand.vue:10]: is this a leftover?

File: packages/components/src/components/brand/brand.scss [hunk 6]
```diff
@@ -1,25 +1,23 @@
...
-	a {
-		@extend %brand-display;
-		color: inherit;
-		// Deactivate underline on site name being made clickable by added to the link as well
-		text-decoration: none;
+	&::before {
+		--db-icon-font-size: #{variables.$db-sizing-md};
+		--db-icon-margin-end: 0;
+		--db-current-icon-color: #{colors.$db-brand-origin-enabled};
 	}
```
Comment (mfranzke) [inline packages/components/src/components/brand/brand.scss:14]: wouldn't we still need this if somebody wraps a link around our element (as it's the new approach)?

Comment (mfranzke) [reply packages/components/src/components/brand/brand.scss:14]: yes, it seems like we do:
image

File: showcases/vue-showcase/src/components/brand/Brand.vue (hunk 5 above)
Comment (nmerget) [reply showcases/vue-showcase/src/components/brand/Brand.vue:10]: yes was a leftover, removed it

File: packages/components/src/components/brand/brand.scss (hunk 6 above)
Comment (nmerget) [reply packages/components/src/components/brand/brand.scss:24]: fixed

File: packages/components/src/components/brand/brand.scss [hunk 7]
```diff
@@ -1,25 +1,33 @@
...
 	font-weight: 700;
 
-	a {
-		@extend %brand-display;
-		color: inherit;
-		// Deactivate underline on site name being made clickable by added to the link as well
-		text-decoration: none;
+	// Deactivate underline on site name being made clickable by added to the link as well
+	text-decoration: none;
+
```
Comment (mfranzke) [inline packages/components/src/components/brand/brand.scss:19]:
```suggestion
```

what for do you need this declaration? It needs to get overwritten on the hyperlink on the outside (only), doesn't it?

File: packages/components/src/components/brand/brand.scss [hunk 8]
```diff
@@ -1,25 +1,33 @@
...
 
-%brand-display {
-	display: flex;
-	gap: variables.$db-spacing-fixed-md;
-	align-items: center;
+:root {
+	* > :has(.db-brand) {
+		// Deactivate underline on site name being made clickable by added to the link as well
+		text-decoration: none;
+	}
```
Comment (mfranzke) [inline packages/components/src/components/brand/brand.scss:8]:
```suggestion
a > :has(.db-brand) {
	// Deactivate underline on site name being made clickable by added to the link as well
	text-decoration: none;
}
```

we could be more restrictive on this one and even also don't need to prefix it by `:root` (or I don't understand the reason).

File: packages/components/src/components/brand/brand.scss (hunk 7 above)
Comment (nmerget) [reply packages/components/src/components/brand/brand.scss:17]: You could add the `<a>` as child into the brand so only the text would be a link. Maybe it's not a use-case, I don't know

Comment (mfranzke) [reply packages/components/src/components/brand/brand.scss]: > You could add the `<a>` as child into the brand so only the text would be a link. Maybe it's not a use-case, I don't know

valid aspect

Comment (mfranzke) [reply packages/components/src/components/brand/brand.scss]: I've added `& a` to `a:has(&)` for that.

**Review by mfranzke — APPROVED** [review]

**Pull request merged**